src/app/components/core/configuration-manager/view-config-item/view-config-item.component.ts

```diff
@@ -9,41 +9,83 @@ import { NgProgress } from '../../../../../../node_modules/ngx-progressbar';
 })
 export class ViewConfigItemComponent implements OnInit {
   @Input() categoryConfigurationData: any;
+  public oldValue;
+  public selectedValue;
+  public isValidJson = true;
 
   constructor(private configService: ConfigurationService,
     private alertService: AlertService,
     public ngProgress: NgProgress) { }
 
   ngOnInit() { }
 
-  public restoreConfigFieldValue(config_item_key: string) {
-    const inputField = <HTMLInputElement>document.getElementById(config_item_key.toLowerCase());
-    inputField.value = inputField.textContent;
-    const cancelButton = <HTMLButtonElement>document.getElementById('btn-cancel-' + config_item_key.toLowerCase());
+  public restoreConfigFieldValue(configItemKey, categoryKey, configValue, configType) {
+    let itemKey = categoryKey.toLowerCase() + '-' + configItemKey.toLowerCase();
+    if (itemKey.includes('select')) {
+      itemKey = itemKey.replace('select-', '');
+    }
+
+    let htmlElement;
+    htmlElement = <HTMLInputElement>document.getElementById(itemKey);
+    if (htmlElement == null) {
+      htmlElement = <HTMLSelectElement>document.getElementById('select-' + itemKey);
+      htmlElement.selectedIndex = (this.selectedValue === 'true' ? 1 : 0);
+    } else {
+      htmlElement.value = htmlElement.textContent;
+    }
+
+    const cancelButton = <HTMLButtonElement>document.getElementById('btn-cancel-' + itemKey);
     cancelButton.classList.add('hidden');
+
+    if (configType.toUpperCase() === 'JSON') {
+      this.isValidJson = this.isValidJsonString(configValue);
+      return;
+    }
   }
 
-  public saveConfigValue(category_name: string, config_item: string, type: string) {
-    const cat_item_id = (category_name.trim() + '-' + config_item.trim()).toLowerCase();
-    const inputField = <HTMLInputElement>document.getElementById(cat_item_id);
-    const value = inputField.value.trim();
-    const id = inputField.id.trim();
-    const cancelButton = <HTMLButtonElement>document.getElementById('btn-cancel-' + id);
+  public saveConfigValue(categoryName, configItem, type) {
+    const catItemId = categoryName.toLowerCase() + '-' + configItem.toLowerCase();
+    let htmlElement;
+    htmlElement = <HTMLInputElement>document.getElementById(catItemId);
```

Comment:
why not `let htmlElement = <HTMLInputElement>document.getElementById(catItemId); ` a single line